### PR TITLE
Add input aspect to definitions

### LINKED ISSUE #1196: Add input aspect to definitions
PR #1189 added an input aspect to a number of rules, because one of its definitions was updated. That made me wonder if it would be helpful to include input aspects in the frontmatter of the definition. This would help authors know what input aspects to use in new rules, and it'd give us a way to check no input aspects needed for the definitions were missing from rules.

### PR DESCRIPTION
**Pull request by Jym77**
Adding input aspects to definitions.
Auto-testing that these aspects are copied in the rules is not part of this PR.

Closes issue(s):

- closes #1196 

Need for Final Call:
This will not require a Final Call (internal meta-data only)

---

### **When creating PR:**

- [X] Make sure you're requesting to **pull a branch** (right side) to the `develop` branch (left side).

### **After creating PR:**

- [X] Add yourself (and co-authors) as "Assignees" for PR.
- [X] Add label to indicate if it's a `Rule`, `Definition` or `Chore`.
- [X] Close the issue that the PR resolves (and make sure the issue is referenced in the top of this comment)
- [X] Optionally request feedback from anyone in particular by assigning them as "Reviewers".

## How to Review And Approve

- Go to the “Files changed” tab
- Here you will have the option to leave comments on different lines.
- Once the review is completed, find the “Review changes” button in the top right, select “Approve” (if you are really confident in the rule) or "Request changes" and click “Submit review”.
- Make sure to also review the proposed Final Call period. In case of disagreement, the longer period wins.

### CODE REVIEW

File: pages/glossary/programmatic-label.md [hunk 1]
```diff
@@ -3,6 +3,8 @@ title: Programmatic Label
 key: programmatic-label
 unambiguous: true
 objective: true
+input_aspects:
+  - DOM tree
```
Comment (WilcoFiers) [inline pages/glossary/programmatic-label.md]: Labelled controls can not be hidden with CSS.

```suggestion
  - CSS styling
  - DOM tree
```

File: pages/glossary/standard-keyboard-navigation.md [hunk 2]
```diff
@@ -3,6 +3,8 @@ title: Standard keyboard navigation
 key: standard-keyboard-navigation
 unambiguous: true
 objective: true
+input_aspects:
+  - DOM tree
```
Comment (WilcoFiers) [inline pages/glossary/standard-keyboard-navigation.md]: I'm not sure how this requires DOM. Might not have any input.
```suggestion
```

File: pages/glossary/visible-text-content.md [hunk 3]
```diff
@@ -3,6 +3,8 @@ title: Visible Text Content
 key: visible-text-content
 unambiguous: true
 objective: true
+input_aspects:
+  - CSS styling
```
Comment (WilcoFiers) [inline pages/glossary/visible-text-content.md]: Talks of node types and attributes, so DOM is needed.

```suggestion
  - CSS styling
  - DOM tree 
```

File: pages/glossary/programmatic-label.md (hunk 1 above)
Comment (Jym77) [reply pages/glossary/programmatic-label.md]: 🤔 The  visibility of the labelled control is currently handled in the rule's applicability (definition is only used in one rule), and the HTML definition of "labelled control" does not speak about visibility.

We could move the visibility to the definition (and require that the programmatic label is visible), but I don't think that it is necessarily. The current definition can be used for "AT specific" labels, which could be useful in other rules.

File: pages/glossary/standard-keyboard-navigation.md (hunk 2 above)
Comment (Jym77) [reply pages/glossary/standard-keyboard-navigation.md]: It needs to know if elements are focusable, so it should include at least the same input aspects.

File: pages/glossary/accessible-name.md [hunk 4]
```diff
@@ -3,6 +3,9 @@ title: Accessible Name
 key: accessible-name
 unambiguous: true
 objective: true
+input_aspects:
+  - Accessibility tree
+  - DOM tree
```
Comment (carlosapaduarte) [inline pages/glossary/accessible-name.md]: CSS is relevant for the accessible name computation.

```suggestion
  - Accessibility tree
  - CSS styling  
  - DOM tree
```

File: pages/glossary/visible.md [hunk 5]
```diff
@@ -3,6 +3,8 @@ title: Visible
 key: visible
 unambiguous: true
 objective: true
+input_aspects:
```
Comment (carlosapaduarte) [inline pages/glossary/visible.md]: I'm not sure we can leave "DOM tree" outside. 
For example, with the `<details><summary>` elements, the details become visible when the `<details>` element gains an `open` attribute.

File: pages/glossary/included-in-the-accessibility-tree.md [hunk 6]
```diff
@@ -3,6 +3,8 @@ title: Included in the accessibility tree
 key: included-in-the-accessibility-tree
 unambiguous: true
 objective: true
+input_aspects:
```
Comment (adilsofficial) [inline pages/glossary/included-in-the-accessibility-tree.md]: Accessibility tree is a subset of the DOM tree. Shouldn't we add DOM tree as well?

Comment (Jym77) [reply pages/glossary/included-in-the-accessibility-tree.md]: I don't think so. The accessibility tree is considered given by whatever UA/AT we're considering (i.e. it is not the job of the rule or the tester to build it).

**Pull request merged**